### PR TITLE
Bugfix v0.14.5 mq connection retries

### PR DESCRIPTION
**Pull request by mattkasun**

### CODE REVIEW

**Review by 0xdcarns — APPROVED** [review]

**Comment by 0xdcarns** [comment]
@theguy951357 rft

**Comment by theguy951357** [comment]
what are my testing steps for this one? am I just looking through the mq logs?

**Comment by 0xdcarns** [comment]
> what are my testing steps for this one? am I just looking through the mq logs?

1) Setup a server as normal
2) Ensure MQ connections is functioning as normal
3) Turn off MQ
4) Make sure it retries connection 3 times, then tries to obtain new certs, by watching logs

**Comment by theguy951357** [comment]
pass

File: netclient/functions/daemon.go [hunk 1]
```diff
@@ -248,22 +248,30 @@ func setupMQTT(cfg *config.ClientConfig, publish bool) (mqtt.Client, error) {
...
+	for count := 0; count < 3; count++ {
+		connecterr = nil
+		if token := client.Connect(); !token.WaitTimeout(30*time.Second) || token.Error() != nil {
+			logger.Log(0, "unable to connect to broker, retrying ...")
+			if token.Error() == nil {
+				connecterr = errors.New("connect timeout")
+			} else {
+				connecterr = token.Error()
+			}
+			if connecterr != nil {
```
Comment (abhishek9686) [inline netclient/functions/daemon.go:261]: isn't this an unnecessary check at this point `connecterr` won't be nil anyways

**Review by abhishek9686 — APPROVED** [review]

**Pull request merged**